### PR TITLE
chore: add the classifier function

### PR DESCRIPTION
**Pull request by aayushmau5**
This PR introduces our classifier function whose job is to get the classification data from the standard object.

The purpose of this PR is to show how the correct path are being generated, so that people could help to make this function better :)

This PR doesn't have any tests and such, which I'm planning to add eventually. Just wanted to get reviews on this function asap.

---

Here's how it works:

here's how our standard object looks like:

```js
const standard = {
    '/servers': {},
    '/servers/*': {},
    '/servers/*/protocol': {}
};
```

Suppose a change has been occured at `/servers/google/protocol`.

Now, running the `classifier` function as `classifier(standard, '/servers/google/protocol');`, 
the flow of generating the correct path for this is as follows:

1. `/servers/google/protocol` gets divided into an array based on `/` as delimiter.
    - `/servers/google/protocol`.split('/') will look like `['', 'servers', 'google', 'protocol']`
2. We start going through this array one element one after another, and joining them with `/`.
    - First, we will take the first element of the array `''`, then join it using `/`. It becomes `''`. We throw out this value at the start.
    - Second, we will take the first two elements, `['', 'servers']`, then join, `/servers`. We look for this `/servers` value in the standard object using `standard['/server']`.
        If found, we continue onto the next step.
    - Third, we will take the frist three elements, `['', 'servers', 'google']`, then join, `/server/google`, we look for this in the standard.
        It is not found. So, here we basically change the last element of the selected array to `*`.
        So, `['', 'servers', '*']`, then we join it, `/servers/*` and look for this key in the standard.
        If found, we go onto the next step. If not found, we now know that `/servers/*` doesn't exist, so we are assuming that the next value `/servers/*/*` doesn't exist. For this reason, we mark the change as `unclassified`.
    - Fourth, we will take the first four elements, but this time, since we have modified the previous array, we will get the modified array, which is `['', 'servers', '*', 'protocol']`. Joining them, we get `/servers/*/protocol`, we get the value. And therefore, this is the desired path.

Would love to get a review on how this implementation is, and how can it improved further.

### CODE REVIEW

File: src/ClassifierHelpers.ts
Comment (derberg) [inline src/ClassifierHelpers.ts:27]: path must be valid json pointer, otherwise you will have a problem when the path points to the channel where channel name contains `/`.

JSON Pointer will make `/` look like `~1`. Example: where channel name is `smartylighting/streetlights/1/0/action/{streetlightId}/turn/off` ([link](https://github.com/asyncapi/parser-js/blob/master/test/wrong/invalid-asyncapi-with-ref-with-line-break.yaml#L64)) -> `/channels/smartylighting~1streetlights~11~10~1action~1{streetlightId}~1turn~1off/parameters/streetlightId/$ref`

There are some helpers in the parser already that might be useful https://github.com/asyncapi/parser-js/blob/master/lib/utils.js

json pointer transformation, tilde/untilde

**Review by derberg — COMMENTED** [review]
Well explained!!!
Some comments for starters

File: src/ClassifierHelpers.ts
Comment (aayushmau5) [reply src/ClassifierHelpers.ts:27]: Yeah, the diffing function we are using [here](https://github.com/asyncapi/diff/pull/18) will escape these path strings. So it's safe to use it here.

Comment (derberg) [reply src/ClassifierHelpers.ts:27]: ah perfect, it outputs json pointer ❤️ 
shouldn't it be explicit here in the jsdoc?

File: src/ClassifierHelpers.ts
Comment (derberg) [inline src/ClassifierHelpers.ts:26]: why standard is any? shouldn't we have an interface for it? or will it follow once standard-related PR is merged?

File: src/ClassifierHelpers.ts
Comment (aayushmau5) [reply src/ClassifierHelpers.ts:27]: oh yeah, will add that in the jsdoc. Thanks :)

File: src/ClassifierHelpers.ts
Comment (aayushmau5) [reply src/ClassifierHelpers.ts:26]: Correct. I'm gonna make related to the standard, and then we will define an interface for it.

**Comment by derberg** [comment]
just fix the code smell reported by sonarcloud and we are ready to merge from my point of view, unless you want more people to have a look (in case of Maciek you would need to wait until next week as he is lazy and took few days off 😆 )

**Comment by aayushmau5** [comment]
wait, I have to add some tests as well. 

But this PR https://github.com/asyncapi/diff/pull/18 is ready to be merged.

**Comment by derberg** [comment]
@aayushmau5 as Maciek reviewed it and blocked from merging I'd like to wait for him. In future remember that when you are ready for another review round, click the button to re-request review

**Review by derberg — DISMISSED** [review]

File: src/classifier.ts
Comment (magicmatatjahu) [inline src/classifier.ts:11]: You can make for it enum type :)

**Review by magicmatatjahu — COMMENTED** [review]
One thing. Overall, awesome! :)

File: src/classifier.ts
Comment (aayushmau5) [reply src/classifier.ts:11]: I will refactor these types when we will move all the types in a `types.ts` file. So, I will convert this to an enum that time. Is it alright?

Comment (magicmatatjahu) [reply src/classifier.ts:11]: Ok, no problem :)

**Review by magicmatatjahu — DISMISSED** [review]
LGTM!

**Comment by aayushmau5** [comment]
fixing the test real quick.

**Review by magicmatatjahu — APPROVED** [review]
LGTM

**Pull request merged**
